src/redux/message-service/saga.ts

```diff
@@ -83,6 +88,7 @@ function* validateMessage(data, signature) {
     action = yield take(actionFilter);
   }
   if (action.type === walletActions.VALIDATION_SUCCESS) {
+    yield put(walletActions.storeMessageRequest(data, signature, Direction.Received));
```

Comment:
I think this should probably be done in the wallet e.g. if `VALIDATION_SUCCESS` comes back, you know that the wallet will have also stored it. I'm pretty sure there's never a case where you receive a valid state and don't want the wallet to save it, so we should remove the burden of remembering to save from the application developer.